src/main/java/au/org/emii/geoserver/extensions/filters/layer/data/Filter.java

```diff
@@ -19,7 +19,7 @@
     private Boolean enabled;
     private Boolean visualised;
     private Boolean excludedFromDownload;
-    public HashMap extras = new HashMap();
+    public Map<String, String> extras = new HashMap<String, String>();
```

Comment:
I wanted to do it too, but there are 2 constructors. Since I'm not a Java expert I decided to stick with what was there already, not being sure if that's the Java way of doing things.

A better idea is to definitely just have it in both constructors, so since you've raised this, I'll just modify it.